src/Autoloader.php

```diff
@@ -36,7 +38,37 @@ public static function init() {
 			return false;
 		}
 
-		return require $autoloader;
+		$autoloader_result = require $autoloader;
+		if ( ! $autoloader_result ) {
+			return false;
+		}
+
+		self::registerPsr4Autoloader();
+
+		return $autoloader_result;
+	}
+
+	/**
+	 * Define a PSR4 autoloader for the dependency injection engine to work.
+	 * Function grabbed from https://container.thephpleague.com/3.x
+	 */
+	protected static function registerPsr4Autoloader() {
```

Comment:
I wasn't sure if this would be temporary or not, just that it was needed. Yes a warning like "verify if that is still needed after that JP PR is merged" would be appropriate.